src/ethereum/shanghai/fork.py

```diff
@@ -635,7 +636,13 @@ def process_transaction(
 
     output = process_message_call(message, env)
 
-    gas_used = tx.gas - output.gas_left
+    floor = tokens_in_calldata * TOTAL_COST_FLOOR_PER_TOKEN + TX_BASE_COST
+    if gas_used + tokens_in_calldata * STANDARD_TOKEN_COST  < floor:
```

Comment:
Would it be more readable if `gas_used + tokens_in_calldata * STANDARD_TOKEN_COST` was stored in an intermediate variable?